### DISCUSSION TITLE
ActualBudget LXC unable to start service

### DISCUSSION THREAD

**Discussion by andrew-op** [tteck/Proxmox#3962] [Help Needed] [answered]
After installing the ActualBudget LXC, I noticed the service was inaccessible by IP. After checking the systemctl status, it shows the following:

actualbudget.service                                              loaded failed failed    Actual Budget Service

I ran `journalctl -u actualbudget.service -f` to try and get more info and it gave me the following:

`Oct 23 20:38:48 actualbudget yarn[2480]:     at new Database (/opt/actualbudget/node_modules/better-sqlite3/lib/database.js:65:9)`
`Oct 23 20:38:48 actualbudget yarn[2480]:     at openDatabase (file:///opt/actualbudget/src/db.js:57:30)`
`Oct 23 20:38:48 actualbudget yarn[2480]:     at getAccountDb (file:///opt/actualbudget/src/account-db.js:12:18`)
`Oct 23 20:38:48 actualbudget yarn[2480]:     at file:///opt/actualbudget/src/app-sync/services/files-service.js:195:39`
`Oct 23 20:38:48 actualbudget yarn[2480]:     at ModuleJob.run (node:internal/modules/esm/module_job:234:25)`
`Oct 23 20:38:48 actualbudget yarn[2480]:     at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)`
`Oct 23 20:38:48 actualbudget yarn[2480]:     at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)`
`Oct 23 20:38:48 actualbudget yarn[2480]: Node.js v20.18.0`
`Oct 23 20:38:48 actualbudget systemd[1]: actualbudget.service: Main process exited, code=exited, status=1/FAILURE`
`Oct 23 20:38:48 actualbudget systemd[1]: actualbudget.service: Failed with result 'exit-code'.`

Im unsure at this point what might be causing this.

**Comment by andrew-op** [comment]
I navigated to /opt/actualbudget and ran the recommended `yarn start` from the Actual Budget docs and was met with the following:

![image](https://github.com/user-attachments/assets/92fe19b0-920b-43eb-b970-6d10f2faa505)

**Comment by andrew-op** [comment, marked as answer]
If anyone runs into this issue in the future, here is what I did to resolve this:

1. First, stop the service:
```bash
systemctl stop actualbudget.service
```

2. Create the required directories and set permissions:
```bash
mkdir -p /opt/actualbudget/server-files
chown -R root:root /opt/actualbudget/server-files
chmod 755 /opt/actualbudget/server-files
```

3. Create a configuration file:
```bash
cat <<EOF > /opt/actualbudget/.env
ACTUAL_UPLOAD_DIR=/opt/actualbudget/server-files
PORT=5006
EOF
```

4. Modify the service file to include the environment file:
```bash
cat <<EOF > /etc/systemd/system/actualbudget.service
[Unit]
Description=Actual Budget Service
After=network.target

[Service]
Type=simple
User=root
Group=root
WorkingDirectory=/opt/actualbudget
EnvironmentFile=/opt/actualbudget/.env
ExecStart=/usr/bin/yarn start
Restart=always
RestartSec=10

[Install]
WantedBy=multi-user.target
EOF
```

5. Reload systemd and restart the service:
```bash
systemctl daemon-reload
systemctl restart actualbudget.service
```

6. Check the status:
```bash
systemctl status actualbudget.service
```

**Reply by tteck** [reply]
Good catch! Something must have changed with a new version.
